core/src/main/java/io/substrait/relation/AbstractJoin.java

```diff
@@ -0,0 +1,9 @@
+package io.substrait.relation;
+
+import io.substrait.expression.Expression;
+import java.util.Optional;
+
+public abstract class AbstractJoin extends BiRel implements HasExtension {
+
+  public abstract Optional<Expression> getPostJoinFilter();
```

Comment:
I just want to add that a NestedLoopJoin does not have a post join filter, so it wouldn't be able to use this abstract class. Maybe it's not needed after all?

Reply:
Ah good point. I will update the PR.